sdk/keyvault/azure-security-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/KeyClientJavaDocCodeSnippets.java

```diff
@@ -488,10 +533,136 @@ public void getRandomBytes() {
+        KeyClient keyClient = createClient();
+        // BEGIN: com.azure.security.keyvault.keys.KeyClient.getKeyRotationPolicy#String
+        KeyRotationPolicy keyRotationPolicy = keyClient.getKeyRotationPolicy("keyName");
+
+        System.out.printf("Retrieved key rotation policy with id: %s%n", keyRotationPolicy.getId());
+        // END: com.azure.security.keyvault.keys.KeyClient.getKeyRotationPolicy#String
+
+        // BEGIN: com.azure.security.keyvault.keys.KeyClient.getKeyRotationPolicyWithResponse#String-Context
+        Response<KeyRotationPolicy> keyRotationPolicyResponse =
+            keyClient.getKeyRotationPolicyWithResponse("keyName", new Context("key1", "value1"));
+
+        System.out.printf("Response received successfully with status code: %d. Retrieved key rotation policy"
+            + "with id: %s%n", keyRotationPolicyResponse.getStatusCode(), keyRotationPolicyResponse.getValue().getId());
+        // END: com.azure.security.keyvault.keys.KeyClient.getKeyRotationPolicyWithResponse#String-Context
+    }
+
+    /**
+     * Generates code samples for using {@link KeyClient#updateKeyRotationPolicy(String, KeyRotationPolicyProperties)}
+     * and {@link KeyClient#updateKeyRotationPolicyWithResponse(String, KeyRotationPolicyProperties, Context)}.
+     */
+    public void updateKeyRotationPolicy() {
+        KeyClient keyClient = createClient();
+        // BEGIN: com.azure.security.keyvault.keys.KeyClient.updateKeyRotationPolicy#String-KeyRotationPolicyProperties
+        List<KeyRotationLifetimeAction> lifetimeActions = new ArrayList<>();
+        KeyRotationLifetimeAction rotateLifetimeAction = new KeyRotationLifetimeAction(KeyRotationPolicyAction.ROTATE)
+            .setTimeAfterCreate("P90D")
+            .setTimeBeforeExpiry("P45D");
+        KeyRotationLifetimeAction notifyLifetimeAction = new KeyRotationLifetimeAction(KeyRotationPolicyAction.NOTIFY)
+            .setTimeAfterCreate("P90D")
+            .setTimeBeforeExpiry("P45D");
+
+        lifetimeActions.add(rotateLifetimeAction);
+        lifetimeActions.add(notifyLifetimeAction);
+
+        KeyRotationPolicyProperties policyProperties = new KeyRotationPolicyProperties()
+            .setLifetimeActions(lifetimeActions)
+            .setExpiryTime("P6M");
+
+        KeyRotationPolicy keyRotationPolicy =
+            keyClient.updateKeyRotationPolicy("keyName", policyProperties);
+
+        System.out.printf("Updated key rotation policy with id: %s%n", keyRotationPolicy.getId());
+        // END: com.azure.security.keyvault.keys.KeyClient.updateKeyRotationPolicy#String-KeyRotationPolicyProperties
+
+        // BEGIN: com.azure.security.keyvault.keys.KeyClient.updateKeyRotationPolicyWithResponse#String-KeyRotationPolicyProperties-Context
+        List<KeyRotationLifetimeAction> myLifetimeActions = new ArrayList<>();
+        KeyRotationLifetimeAction myRotateLifetimeAction = new KeyRotationLifetimeAction(KeyRotationPolicyAction.ROTATE)
```

Comment:
Do we not use builders for this? `new KeyRotationLifetimeAction.Builder().setTimeAfterCreate("P90D").build()`?

Again, just asking so I can better learn how we Java

Reply:
We don't use builders for models such as this, but we do use a fluent pattern for all of them so we can chain set operations, for example:

``` java
ModelClass myModel= new ModelClass(requiredPoperty1, requiredproperty2)
    .setOptionalProperty("some-value");
    .setOtherOptionalProperty("some-value");
```